### DISCUSSION TITLE
Custom error handling.

### DISCUSSION THREAD

**Discussion by ajakam** [d3/d3#3901] [Q&A] [answered]
We have an application that renders interactive charts with D3. This application has error handling, monitoring & alerting that notifies us if any unexpected errors occur.

We encountered an issue recently where errors logged by D3 weren't present in the monitoring system.

![image](https://github.com/user-attachments/assets/6d7aee76-a14a-4daa-b780-4502f6ece006)

After a bit of digging, it looks like D3 is catching & logging these errors, but isn't re-throwing them.

I've looked in the documentation but haven't found anything specific around global D3 error handling.

Is there any way to change/configure this behaviour so that errors are re-thrown, or not caught by D3 at all?

**Comment by Fil** [comment, marked as answer]
It's the browser that's doing the logging, not D3. D3 does not know (nor needs to know) whether the transform attribute is valid.

**Reply by ajakam** [reply]
Ah - okay, that makes more sense.

So it's the native [setAttribute](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute) where that's being logged - and it only throws if the attribute name is invalid, not the value.

Thanks 👍
